mkfs/mkfs.c

```diff
@@ -17,13 +17,40 @@
 #include <errno.h>
 #include <math.h>
 #include <locale.h>
+#include <time.h>
 
 #include "exfat_ondisk.h"
 #include "libexfat.h"
 #include "mkfs.h"
 
 struct exfat_mkfs_info finfo;
 
+/* random serial generator based on current date & time */
+static unsigned int get_new_serial(void)
+{
+	struct tm *tm;
+	struct timespec ts;
+	time_t now;
+	unsigned int x;
+	unsigned int nsec_u, nsec_l;
+
+	if (clock_gettime(CLOCK_REALTIME, &ts)) {
+		/* set specific time on error */
+		ts.tv_sec = (time_t)1583130092;
+		ts.tv_nsec = 0;
+	}
+	now = ts.tv_sec;
+	tm = localtime(&now);
+
+	nsec_u = (unsigned int)(ts.tv_nsec >> 16);
+	nsec_l = (unsigned int)(ts.tv_nsec & 0xffff);
+
+	x = (nsec_u + tm->tm_hour + tm->tm_min + tm->tm_sec) << 16 |
+	    (nsec_l + (tm->tm_year+1900) + (tm->tm_mon+1) + tm->tm_mday);
```

Comment:
I think that adding nano seconds to the upper and lower 16-bit makes the serial not distinguishable.  because the maximum of 16-bit nano seconds is just 0.065536 (65565 / 1000000) milli seconds.

Reply:
Hmm, something like this case, right?
nsec = 0x3B8AFFFF, tm_sec : 10
after then,
nsec = 0x3B7AFFFF, tm_sec : 26

Yes, can be duplicated. 
How about this?

x = (nsec_u | (tm->tm_hour + tm->tm_min + tm->tm_sec)) << 16 |
        (nsec_l | ((tm->tm_year+1900) + (tm->tm_mon+1) + tm->tm_mday));

or any other good idea?

Reply:
Simpler way. PTAL.